compiler/modules/nback.llvm/llvm.bal

```diff
@@ -235,9 +235,18 @@ public class FunctionDefn {
         return label;
     }
 
-    function genReg() returns string {
-        string reg = "%" + "R" + self.varCount.toString();
-        self.varCount += 1;
+    function genReg(string? name = ()) returns string {
+        string regName = "R" + self.variableNames.length().toString();
```

Comment:
If we are now using user-defined names, then `R` is no longer a choice (it looks too much like a user variable). Something like `_` would be better if it works.

Reply:
Fixed